libs/sdk-ui-kit/src/Form/Input.tsx

```diff
@@ -0,0 +1,72 @@
+// (C) 2020 GoodData Corporation
+import React from "react";
+import isNumber from "lodash/isNumber";
+import isString from "lodash/isString";
+import isNaN from "lodash/isNaN";
+
+import { InputPure, InputPureProps } from "./InputPure";
+
+const isValidNumber = (value: string | number) => isNumber(value) && !isNaN(value);
+const isNumberOrString = (value: string | number) =>
+    isValidNumber(value) || (isString(value) && value.length);
+const toValidValue = (value: string | number) => (isNumberOrString(value) ? value : "");
+
+/**
+ * @internal
+ */
+export interface InputState {
+    value: string | number;
+}
+
+/**
+ * @internal
+ */
+export class Input extends React.PureComponent<InputPureProps, InputState> {
+    static defaultProps = {
+        ...InputPure.defaultProps,
+    };
+    public inputNodeRef: InputPure;
```

Comment:
Can we please use `createRef`? The callback refs were deprecated and this would be a good opportunity to make this change (it was too risky for goodstrap).

Reply:
these refs are leaking and used outside the component